### DISCUSSION TITLE
DebugAI command becomes blurry if your sub is moving down fast

### DISCUSSION THREAD

**Discussion by iridescent-lightning** [FakeFishGames/Barotrauma#14815] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

![20241012091418_1](https://github.com/user-attachments/assets/ea5e7f24-956e-4901-93af-bd4d01ba905f)
opening debugai then drill some holes to let the sub sink. Now the words becomes blurry. I can't really screen shot it because I will get the unbluryable frame.

### Reproduction steps

1. sub editor
2. console command, debugai
3. sink the sub
4. watch the words get blury.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
Did you mean chromatically aberrated?

**Reply by Regalis11** [reply]
Probably not, because chromatic aberration is not applied on these texts or any other UI element.

**Comment by Ek-Videogames** [comment]
This would actually be due to... well I don't actually know the technical term for this artifact, 
but basically the text's position stuttering a little bit every frame because of weird desync position updating/frequency.

- Because (assuming the game is running at a good framerate), the UI is redrawing every time so fast that your perception just averages it out to look like the UI is blurred, when in reality it's just the UI appearing in slightly different spots each frame along that position 'margin of error' relative to the direction of velocity.

The same sort of thing happens with normal UI labels that draw from tooltips when hovering over selected objects in regular gameplay too, if the sub is going fast enough.

Also I did notice something else:
> chromatic aberration is not applied on these texts or any other UI element.

Now that you say that, I actually never noticed this before.
Explain please?
![image](https://github.com/user-attachments/assets/ee823666-f731-4139-81cb-13cc7272bc74)

**Reply by SomeRandomNoobKekeke** [reply]
![Screenshot_3](https://github.com/user-attachments/assets/2891505a-bf2f-487c-9da6-76b92c211e28)
![barodev small](https://github.com/user-attachments/assets/0c0a30ec-08af-4bd5-be11-860b798ca5d8)

**Reply by Regalis11** [reply]
> Explain please?

These debug texts on hulls are not technically part of the UI - they're drawn as a part of the hull, so in a sense it's "a part of the submarine / game world" and gets aberrated like everything else.

**Comment by SomeRandomNoobKekeke** [comment]
On my end they are not blurred, they are shaking

https://github.com/user-attachments/assets/0de4e9ce-eb1a-4a00-afc4-3f578153cf73

Also don't debug while drunk xd
![20241013135126_1](https://github.com/user-attachments/assets/919de4a3-8067-4600-8d6d-35d19a8938c4)

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly, fix coming in the next update. Closing discussion.
